src/metabase/driver.clj

```diff
@@ -1323,6 +1328,60 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti string->val
+  "Parse a string value for insertion based on driver and column definition.
+
+  Takes:
```

Comment:
When I see this I am not sure I'd know what I would need to do without reading python/transforms and figuring out how these strings come about.

My mental model: Implementors should expect string encodings are fixed regardless of driver, e.g. arrays are jsons, numerics have maybe one or more notation (exponent etc, decimal etc), datetimes have some iso format. Intervals have some iso format.

Should we document what kinds of strings drivers are expected to encounter and so they can decide which ones need to be specialised for (as mentioned some drivers might accept the strings as-is)?

Maybe there are some other docs that fix this that I'm not aware of?